core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -351,6 +416,29 @@ private void tryStart() {
         }
     }
 
+    @SneakyThrows(JsonProcessingException.class)
+    final String hash(CreateContainerCmd createCommand) {
+        // TODO add Testcontainers' version to the hash
+        String commandJson = new ObjectMapper()
+            .enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY)
+            .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+            .writeValueAsString(createCommand);
+
+        return DigestUtils.sha1Hex(commandJson);
+    }
+
+    Optional<String> findContainerForReuse(String hash) {
```

Comment:
`private` ?

Reply:
left internal for future testing (and this method will require **a lot** of testing at stage 2 😅)

Reply:
yeah, was assuming something like this.
We sometimes used `@VisibleForTesting` annotation.

Reply:
added 👍